test/annotation_utils.t

```diff
@@ -864,6 +873,170 @@ subtest '_run_rast_workflow_ann' => sub {
 };
 
 
+## Just print to explore the finalgenome data structure and find the contig_ids##
+subtest '_check_contigID_mapping' => sub {
+    my $gn_pairs = [
+        {gn => $ann_genome01, rd_ref => \%rast_details01},
+        {gn => $ann_genome02, rd_ref => \%rast_details02},
+        {gn => $ann_genome1, rd_ref => \%rast_details1},
+        {gn => $ann_genome2, rd_ref => \%rast_details2}
+    ];
+    my $cnt_limit = 5;
+    for my $p (@$gn_pairs) {
+        my %rd = %{ $p->{rd_ref} };
+        my $pgn = $p->{gn};
+        my $cnt = 0;
+        print "\n=========before remapping the rasted genome on $pgn->{id}:\n";
+        print "\n------ keys of the rasted genome on $pgn->{id}:\n".
+               Dumper(keys %{ $pgn });
+        if( $pgn->{contig_ids} &&  @{$pgn->{contig_ids}} > 0) {
+            print "\n------the rasted genome's contig_ids:\n".
+                   Dumper( $pgn->{contig_ids});
+        }
+        if( $pgn->{contigs} &&  @{$pgn->{contigs}} > 0) {
+            for my $ctg (@{$pgn->{contigs}}) {
+                print "\n------the rasted genome contig's id=$ctg->{id}";
+            }
+        }
+        if( $pgn->{features} && @{$pgn->{features}} > 0) {
+            $cnt = 0;
+            for my $ftr (@{$pgn->{features}}) {
+                $cnt += 1;
+                print "\n------the rasted genome feature location's contig_id=".
+                      "$ftr->{location}[0][0]";
+                if( $cnt > $cnt_limit ) {
+                    last;
+                }
+            }
+        }
+        if( $pgn->{non_coding_features} &&
+            @{$pgn->{non_coding_features}} > 0) {
+            $cnt = 0;
+            for my $nc_ftr (@{$pgn->{non_coding_features}}) {
+                $cnt += 1;
+                print "\n------the rasted genome nc_feature location's contig_id=".
+                      "$nc_ftr->{location}[0][0]";
+                if( $cnt > $cnt_limit ) {
+                    last;
+                }
+            }
+        }
```

Comment:
This code snippet is repeated for four different keys (`features`, `non_coding_features`, `cdss`, and `mrnas`. As soon as you write out the same pattern more than once, you should be asking yourself if it could be refactored.

Reply:
I did think of that.  Now I carved it out.